### ISSUE TITLE
I get a lot of undefined CONFIG_... errros (IDFGH-11556)

### ISSUE THREAD

**Issue by shlomo-edu** [espressif/esp-idf#12677]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### IDF version.

v5.3-dev-491-g30eb2918c6

### Operating System used.

Windows

### How did you build your project?

Command line with Make

### If you are using Windows, please specify command line type.

None

### What is the expected behavior?

I clone the project.
I am trying to compile the BLE gatt_server example
1. I see in Visual Studio Code that all the include files from "freetos" and below, are all marked with red underline (which means that the it can't find them).
When I try to compile it (with 
I get errors.
The fits error is: unknown type name 'esp_bluedroid_config_t'
 on this line:
`esp_bluedroid_config_t bluedroid_cfg = BT_BLUEDROID_INIT_CONFIG_DEFAULT();
`
I also tried to copy the code (the C file) to another project that I did manage to compile (HID) and there I didn't have the problem with the red line (it found the include files), but I had several undeclared CONFIG_ variable such as:
CONFIG_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF

### What is the actual behavior?

1. I should find the include files.
2. It should compile.

### Steps to reproduce.

1. clone the ESP-IDF
2. Open Visual Sudio Clode
2. Open Folder: C:\ESP32-BLE\esp-idf\examples\bluetooth\bluedroid\ble\gatt_server
3. Click on: View->Command Palette...
4. Select: ESP-IDF: Set Espressif device rarget
5. Click on: View->Command Palette...
6: Select: ESP-IDF: Build, Flush, Start...

### Build or installation Logs.

```plain
[18/32] Building C object esp-idf/main/CMakeFiles/__idf_main.dir/ble_server.c.obj
FAILED: esp-idf/main/CMakeFiles/__idf_main.dir/ble_server.c.obj 
C:\Users\shlomoa\.espressif\tools\xtensa-esp32s3-elf\esp-2022r1-11.2.0\xtensa-esp32s3-elf\bin\xtensa-esp32s3-elf-gcc.exe -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" -DSOC_MMU_PAGE_SIZE=CONFIG_MMU_PAGE_SIZE -DUNITY_INCLUDE_CONFIG_H -IC:/ESP32-WROOM/hid/build/config -IC:/ESP32-WROOM/hid/main -IC:/Users/shlomoa/esp/esp-idf/components/newlib/platform_include -IC:/Users/shlomoa/esp/esp-idf/components/freertos/FreeRTOS-Kernel/include -IC:/Users/shlomoa/esp/esp-idf/components/freertos/esp_additions/include/freertos -IC:/Users/shlomoa/esp/esp-idf/components/freertos/FreeRTOS-Kernel/portable/xtensa/include -IC:/Users/shlomoa/esp/esp-idf/components/freertos/esp_additions/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_hw_support/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_hw_support/include/soc -IC:/Users/shlomoa/esp/esp-idf/components/esp_hw_support/include/soc/esp32s3 -IC:/Users/shlomoa/esp/esp-idf/components/esp_hw_support/port/esp32s3/. -IC:/Users/shlomoa/esp/esp-idf/components/esp_hw_support/port/esp32s3/private_include -IC:/Users/shlomoa/esp/esp-idf/components/heap/include -IC:/Users/shlomoa/esp/esp-idf/components/log/include -IC:/Users/shlomoa/esp/esp-idf/components/soc/include -IC:/Users/shlomoa/esp/esp-idf/components/soc/esp32s3/. -IC:/Users/shlomoa/esp/esp-idf/components/soc/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/hal/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/hal/include -IC:/Users/shlomoa/esp/esp-idf/components/hal/platform_port/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_rom/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_rom/include/esp32s3 -IC:/Users/shlomoa/esp/esp-idf/components/esp_rom/esp32s3 -IC:/Users/shlomoa/esp/esp-idf/components/esp_common/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_system/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_system/port/soc -IC:/Users/shlomoa/esp/esp-idf/components/esp_system/port/include/private 
-IC:/Users/shlomoa/esp/esp-idf/components/xtensa/include -IC:/Users/shlomoa/esp/esp-idf/components/xtensa/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/lwip/include -IC:/Users/shlomoa/esp/esp-idf/components/lwip/include/apps -IC:/Users/shlomoa/esp/esp-idf/components/lwip/include/apps/sntp -IC:/Users/shlomoa/esp/esp-idf/components/lwip/lwip/src/include -IC:/Users/shlomoa/esp/esp-idf/components/lwip/port/esp32/include -IC:/Users/shlomoa/esp/esp-idf/components/lwip/port/esp32/include/arch -IC:/Users/shlomoa/esp/esp-idf/components/esp_ringbuf/include -IC:/Users/shlomoa/esp/esp-idf/components/efuse/include -IC:/Users/shlomoa/esp/esp-idf/components/efuse/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/driver/include -IC:/Users/shlomoa/esp/esp-idf/components/driver/deprecated -IC:/Users/shlomoa/esp/esp-idf/components/driver/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_pm/include -IC:/Users/shlomoa/esp/esp-idf/components/mbedtls/port/include -IC:/Users/shlomoa/esp/esp-idf/components/mbedtls/mbedtls/include -IC:/Users/shlomoa/esp/esp-idf/components/mbedtls/mbedtls/library -IC:/Users/shlomoa/esp/esp-idf/components/mbedtls/esp_crt_bundle/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_app_format/include -IC:/Users/shlomoa/esp/esp-idf/components/bootloader_support/include -IC:/Users/shlomoa/esp/esp-idf/components/bootloader_support/bootloader_flash/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_partition/include -IC:/Users/shlomoa/esp/esp-idf/components/app_update/include -IC:/Users/shlomoa/esp/esp-idf/components/spi_flash/include -IC:/Users/shlomoa/esp/esp-idf/components/pthread/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_timer/include -IC:/Users/shlomoa/esp/esp-idf/components/app_trace/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_event/include -IC:/Users/shlomoa/esp/esp-idf/components/nvs_flash/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_phy/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_phy/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/vfs/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_netif/include -IC:/Users/shlomoa/esp/esp-idf/components/wpa_supplicant/include -IC:/Users/shlomoa/esp/esp-idf/components/wpa_supplicant/port/include -IC:/Users/shlomoa/esp/esp-idf/components/wpa_supplicant/esp_supplicant/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_wifi/include -IC:/Users/shlomoa/esp/esp-idf/components/unity/include -IC:/Users/shlomoa/esp/esp-idf/components/unity/unity/src -IC:/Users/shlomoa/esp/esp-idf/components/cmock/CMock/src -IC:/Users/shlomoa/esp/esp-idf/components/console -IC:/Users/shlomoa/esp/esp-idf/components/http_parser -IC:/Users/shlomoa/esp/esp-idf/components/esp-tls -IC:/Users/shlomoa/esp/esp-idf/components/esp-tls/esp-tls-crypto -IC:/Users/shlomoa/esp/esp-idf/components/esp_adc/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_adc/interface -IC:/Users/shlomoa/esp/esp-idf/components/esp_adc/esp32s3/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_adc/deprecated/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_eth/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_gdbstub/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_gdbstub/xtensa -IC:/Users/shlomoa/esp/esp-idf/components/esp_gdbstub/esp32s3 -IC:/Users/shlomoa/esp/esp-idf/components/esp_hid/include -IC:/Users/shlomoa/esp/esp-idf/components/tcp_transport/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_http_client/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_http_server/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_https_ota/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_lcd/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_lcd/interface -IC:/Users/shlomoa/esp/esp-idf/components/protobuf-c/protobuf-c -IC:/Users/shlomoa/esp/esp-idf/components/protocomm/include/common -IC:/Users/shlomoa/esp/esp-idf/components/protocomm/include/security -IC:/Users/shlomoa/esp/esp-idf/components/protocomm/include/transports -IC:/Users/shlomoa/esp/esp-idf/components/esp_local_ctrl/include -IC:/Users/shlomoa/esp/esp-idf/components/esp_psram/include -IC:/Users/shlomoa/esp/esp-idf/components/espcoredump/include -IC:/Users/shlomoa/esp/esp-idf/components/espcoredump/include/port/xtensa -IC:/Users/shlomoa/esp/esp-idf/components/wear_levelling/include -IC:/Users/shlomoa/esp/esp-idf/components/sdmmc/include -IC:/Users/shlomoa/esp/esp-idf/components/fatfs/diskio -IC:/Users/shlomoa/esp/esp-idf/components/fatfs/vfs -IC:/Users/shlomoa/esp/esp-idf/components/fatfs/src -IC:/Users/shlomoa/esp/esp-idf/components/idf_test/include -IC:/Users/shlomoa/esp/esp-idf/components/idf_test/include/esp32s3 -IC:/Users/shlomoa/esp/esp-idf/components/ieee802154/include -IC:/Users/shlomoa/esp/esp-idf/components/json/cJSON -IC:/Users/shlomoa/esp/esp-idf/components/mqtt/esp-mqtt/include -IC:/Users/shlomoa/esp/esp-idf/components/perfmon/include -IC:/Users/shlomoa/esp/esp-idf/components/spiffs/include -IC:/Users/shlomoa/esp/esp-idf/components/touch_element/include -IC:/Users/shlomoa/esp/esp-idf/components/ulp/ulp_common/include -IC:/Users/shlomoa/esp/esp-idf/components/ulp/ulp_common/include/esp32s3 -IC:/Users/shlomoa/esp/esp-idf/components/usb/include -IC:/Users/shlomoa/esp/esp-idf/components/wifi_provisioning/include -IC:/ESP32-WROOM/hid/components/hid/include -mlongcalls  -fdiagnostics-color=always -ffunction-sections -fdata-sections 
-Wall -Werror=all -Wno-error=unused-function -Wno-error=unused-variable -Wno-error=deprecated-declarations -Wextra 
-Wno-unused-parameter -Wno-sign-compare -Wno-enum-conversion -gdwarf-4 -ggdb -Og -fmacro-prefix-map=C:/ESP32-WROOM/hid=. -fmacro-prefix-map=C:/Users/shlomoa/esp/esp-idf=/IDF -fstrict-volatile-bitfields -Wno-error=unused-but-set-variable -fno-jump-tables -fno-tree-switch-conversion -DconfigENABLE_FREERTOS_DEBUG_OCDAWARE=1 -std=gnu17 -Wno-old-style-declaration -D_GNU_SOURCE -DIDF_VER=\"v5.0.2-dirty\" -DESP_PLATFORM -D_POSIX_READER_WRITER_LOCKS -MD -MT esp-idf/main/CMakeFiles/__idf_main.dir/ble_server.c.obj -MF esp-idf\main\CMakeFiles\__idf_main.dir\ble_server.c.obj.d -o 
esp-idf/main/CMakeFiles/__idf_main.dir/ble_server.c.obj -c C:/ESP32-WROOM/hid/main/ble_server.c
In file included from C:/ESP32-WROOM/hid/components/hid/include/esp_bt.h:17,
                 from C:/ESP32-WROOM/hid/main/ble_server.c:28:
C:/ESP32-WROOM/hid/main/ble_server.c: In function 'ble_server':
C:/ESP32-WROOM/hid/components/hid/include/esp_bt_cfg.h:120:57: error: 'CONFIG_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF' 
undeclared (first use in this function); did you mean 'DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF'?
  120 |     #define DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF CONFIG_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/ESP32-WROOM/hid/components/hid/include/esp_bt_cfg.h:120:57: note: in definition of macro 'DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF'
  120 |     #define DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF CONFIG_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/ESP32-WROOM/hid/main/ble_server.c:697:41: note: in expansion of macro 'BT_CONTROLLER_INIT_CONFIG_DEFAULT'       
  697 |     esp_bt_controller_config_t bt_cfg = BT_CONTROLLER_INIT_CONFIG_DEFAULT();
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/ESP32-WROOM/hid/components/hid/include/esp_bt_cfg.h:120:57: note: each undeclared identifier is reported only once for each function it appears in
  120 |     #define DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF CONFIG_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/ESP32-WROOM/hid/components/hid/include/esp_bt_cfg.h:120:57: note: in definition of macro 'DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF'
  120 |     #define DEFAULT_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF CONFIG_BT_LE_COEX_PHY_CODED_TX_RX_TLIM_EFF
      |                                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
C:/ESP32-WROOM/hid/main/ble_server.c:697:41: note: in expansion of macro 'BT_CONTROLLER_INIT_CONFIG_DEFAULT'       
  697 |     esp_bt_controller_config_t bt_cfg = BT_CONTROLLER_INIT_CONFIG_DEFAULT();
      |                                         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[23/32] Performing build step for 'bootloader'
[1/106] Building C object esp-idf/xtensa/CMakeFiles/__idf_xtensa.dir/eri.c.obj
[2/106] Generating project_elf_src_esp32s3.c
[3/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/sdm_periph.c.obj
[4/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/i2s_periph.c.obj
[5/106] Building C object esp-idf/xtensa/CMakeFiles/__idf_xtensa.dir/xt_trax.c.obj
[6/106] Building C object CMakeFiles/bootloader.elf.dir/project_elf_src_esp32s3.c.obj
[7/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/adc_periph.c.obj
[8/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/gpio_periph.c.obj
[9/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/i2c_periph.c.obj
[10/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/gdma_periph.c.obj
[11/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/mcpwm_periph.c.obj
[12/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/interrupts.c.obj
[13/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/pcnt_periph.c.obj
[14/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/dport_access_common.c.obj
[15/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/lldesc.c.obj
[16/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/dedic_gpio_periph.c.obj
[17/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/lcd_periph.c.obj
[18/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/ledc_periph.c.obj
[19/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/rmt_periph.c.obj
[20/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/timer_periph.c.obj
[21/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/sdio_slave_periph.c.obj
[22/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/spi_periph.c.obj
[23/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/rtc_io_periph.c.obj
[24/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/sdmmc_periph.c.obj
[25/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/touch_sensor_periph.c.obj
[26/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/usb_otg_periph.c.obj
[27/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/uart_periph.c.obj
[28/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/usb_periph.c.obj
[29/106] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/mmu_hal.c.obj
[30/106] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/efuse_hal.c.obj
[31/106] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/esp32s3/efuse_hal.c.obj
[32/106] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/cache_hal.c.obj
[33/106] Building C object esp-idf/hal/CMakeFiles/__idf_hal.dir/mpu_hal.c.obj
[34/106] Building C object esp-idf/soc/CMakeFiles/__idf_soc.dir/esp32s3/temperature_sensor_periph.c.obj
[35/106] Building C object esp-idf/esp_app_format/CMakeFiles/__idf_esp_app_format.dir/esp_app_desc.c.obj
[36/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_clock_init.c.obj
[37/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_mem.c.obj
[38/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_partitions.c.obj
[39/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/secure_boot.c.obj[40/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/bootloader_flash/src/flash_qio_mode.c.obj
[41/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_common.c.obj
[42/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_common_loader.c.obj
[43/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/bootloader_flash/src/bootloader_flash.c.obj
[44/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/bootloader_flash/src/bootloader_flash_config_esp32s3.c.obj
[45/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_random.c.obj
[46/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_efuse.c.obj
[47/106] Building C object esp-idf/micro-ecc/CMakeFiles/__idf_micro-ecc.dir/uECC_verify_antifault.c.obj
[48/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_random_esp32s3.c.obj
[49/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32s3/bootloader_soc.c.obj
[50/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_console_loader.c.obj
[51/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_clock_loader.c.obj
[52/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/flash_encrypt.c.obj
[53/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32s3/bootloader_sha.c.obj
[54/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp_image_format.c.obj
[55/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_utility.c.obj
[56/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/esp32s3/bootloader_esp32s3.c.obj
[57/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_console.c.obj
[58/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/esp_memory_utils.c.obj       
[59/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_panic.c.obj
[60/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/rtc_clk_init.c.obj
[61/106] Building C object esp-idf/bootloader_support/CMakeFiles/__idf_bootloader_support.dir/src/bootloader_init.c.obj
[62/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32s3/esp_efuse_utility.c.obj
[63/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32s3/esp_efuse_fields.c.obj
[64/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/rtc_clk.c.obj   
[65/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32s3/esp_efuse_table.c.obj
[66/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/esp32s3/esp_efuse_rtc_calib.c.obj
[67/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/rtc_init.c.obj  
[68/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_api.c.obj
[69/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/efuse_controller/keys/with_key_purposes/esp_efuse_api_key.c.obj
[70/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_fields.c.obj
[71/106] Building C object esp-idf/esp_system/CMakeFiles/__idf_esp_system.dir/esp_err.c.obj
[72/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/chip_info.c.obj 
[73/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/rtc_pm.c.obj    
[74/106] Building C object esp-idf/efuse/CMakeFiles/__idf_efuse.dir/src/esp_efuse_utility.c.obj
[75/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/rtc_time.c.obj  
[76/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/cpu.c.obj
[77/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_sys.c.obj
[78/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_crc.c.obj
[79/106] Building C object esp-idf/esp_hw_support/CMakeFiles/__idf_esp_hw_support.dir/port/esp32s3/rtc_sleep.c.obj 
[80/106] Building C object esp-idf/esp_common/CMakeFiles/__idf_esp_common.dir/src/esp_err_to_name.c.obj
[81/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_regi2c.c.obj
[82/106] Building ASM object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_longjmp.S.obj
[83/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_uart.c.obj
[84/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_systimer.c.obj
[85/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_spiflash.c.obj
[86/106] Building C object esp-idf/esp_rom/CMakeFiles/__idf_esp_rom.dir/patches/esp_rom_efuse.c.obj
[87/106] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log_noos.c.obj
[88/106] Building C object esp-idf/main/CMakeFiles/__idf_main.dir/bootloader_start.c.obj
[89/106] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log.c.obj
[90/106] Building C object esp-idf/log/CMakeFiles/__idf_log.dir/log_buffers.c.obj
[91/106] Linking C static library esp-idf\log\liblog.a
[92/106] Linking C static library esp-idf\esp_rom\libesp_rom.a
[93/106] Linking C static library esp-idf\esp_common\libesp_common.a
[94/106] Linking C static library esp-idf\esp_hw_support\libesp_hw_support.a
[95/106] Linking C static library esp-idf\esp_system\libesp_system.a
[96/106] Linking C static library esp-idf\efuse\libefuse.a
[97/106] Linking C static library esp-idf\bootloader_support\libbootloader_support.a
[98/106] Linking C static library esp-idf\esp_app_format\libesp_app_format.a
[99/106] Linking C static library esp-idf\hal\libhal.a
[100/106] Linking C static library esp-idf\micro-ecc\libmicro-ecc.a
[101/106] Linking C static library esp-idf\soc\libsoc.a
[102/106] Linking C static library esp-idf\xtensa\libxtensa.a
[103/106] Linking C static library esp-idf\main\libmain.a
[104/106] Linking C executable bootloader.elf
[105/106] Generating binary image from built executable
esptool.py v4.5.1
Creating esp32s3 image...
Merged 1 ELF section
Successfully created esp32s3 image.
Generated C:/ESP32-WROOM/hid/build/bootloader/bootloader.bin
[106/106] cmd.exe /C "cd /D C:\ESP32-WROOM\hid\build\bootloader\esp-idf\esptool_py && C:\Users\shlomoa\.espressif\python_env\idf5.0_py3.8_env\Scripts\python.exe C:/Users/shlomoa/esp/esp-idf/components/partition_table/check_sizes.py --offset 0x8000 bootloader 0x0 C:/ESP32-WROOM/hid/build/bootloader/bootloader.bin"
Bootloader binary size 0x5030 bytes. 0x2fd0 bytes (37%) free.
ninja: build stopped: subcommand failed.

 *  The terminal process "C:\WINDOWS\System32\WindowsPowerShell\v1.0\powershell.exe -Command ninja " terminated with exit code: 1.
```

**Comment by Halley826** [comment]
Please use the following command to update the submodules: git submodule update.
I have tested using the commit you provided and it is functioning properly:
![image](https://github.com/espressif/esp-idf/assets/92585905/fc6c2174-204c-4413-9c2d-ad138dc53679)

**Comment by zhp0406** [comment]
@shlomo-edu 
Have you checked if the issue still exists after using 'git submodule update'?

**Comment by shlomo-edu** [comment]
It probly solved the unresolved issues, but now I get the following error:
C:/Users/shlomoa/esp/esp-idf/components/xtensa/include/xtensa/xtensa_api.h:104:1: error: expected '=', ',', ';', 'asm' or '__attribute__' before '{' token
  104 | {
      | ^
line 104 is the { in this code:
```
/*
-------------------------------------------------------------------------------
  Call this function to set the specified (s/w) interrupt.
-------------------------------------------------------------------------------
*/
static inline void xt_set_intset(unsigned int arg)
{
    xthal_set_intset(arg);
}

```
And then the rest of the file is full of errors, so I assume that the problem is:
static inline void xt_set_intset(unsigned int arg)
but I don't know what?

**Comment by zhp0406** [comment]
The error still exists within the components, indicating that not all updates to the components have taken effect.

**Comment by zhp0406** [comment]
For the installation of the tool, you can take a look at the documentation: https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/windows-setup.html

At this point, none of the examples should be buildable. Not only the BLE examples, because your environment has not been set up yet.

**Comment by shlomo-edu** [comment]
I downloaded the: Universal Online Installer 2.23 Windows 10, 11 Size: 4 MB
From: https://dl.espressif.com/dl/esp-idf/?idf=4.4
I installed it. It asked me where to install it, and I gave the path: to C:\ESP-NEW
But, it installed something in C:\ESP-NEW and some other stuff in: C:\Espressif
See the attached picture.
At the end it displayed a message to build: idf.py build
So I tried to do it in: C:\ESP-NEW, but it doesn't have CMakeLists.txt
So I tried to compile in: C:\Espressif\frameworks\esp-idf-v5.1.2
But now I receive the following errors:
1. -- Detecting CXX compiler ABI info - failed
2. The C++ compiler   "C:/MinGW/bin/mingw32-gcc.exe"  is not able to compile a simple test program.
3. mingw32-gcc.exe: error: CreateProcess: No such file or directory
    ninja: build stopped: subcommand failed.  
  
![image](https://github.com/espressif/esp-idf/assets/83693785/2131944e-11da-4c29-9879-fe98e53f37fa)
